### PR TITLE
Send telemetry event with new project/function details

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**

### CODE REVIEW

File: src/commands/createFunction.ts [hunk 1]
```diff
@@ -182,4 +184,8 @@ export async function createFunction(
 
     const newFileUri: vscode.Uri = vscode.Uri.file(newFilePath);
     vscode.window.showTextDocument(await vscode.workspace.openTextDocument(newFileUri));
+
+    if (telemetryReporter) {
```
Comment (StephenWeatherford) [inline src/commands/createFunction.ts]: Would it be better to hang this onto the main azureFunctions.createFunction/createProject events?  We'd need a way to do that, though, so may not be worth it.

File: src/commands/createNewProject.ts [hunk 2]
```diff
@@ -204,7 +204,7 @@ async function createJavaFunctionProject(outputChannel: OutputChannel, functionA
     return appName;
 }
 
-export async function createNewProject(outputChannel: OutputChannel, functionAppPath?: string, openFolder: boolean = true, ui: IUserInterface = new VSCodeUI()): Promise<void> {
+export async function createNewProject(outputChannel: OutputChannel, telemetryReporter: TelemetryReporter | undefined, functionAppPath?: string, openFolder: boolean = true, ui: IUserInterface = new VSCodeUI()): Promise<void> {
```
Comment (nturinski) [inline src/commands/createNewProject.ts]: Why not just make this optional?

File: src/commands/createFunction.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply src/commands/createFunction.ts]: Refactored to do just that - let me know what you think

File: src/commands/createNewProject.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply src/commands/createNewProject.ts]: I refactored this anyways, but there's a difference between 'optional' parameters and 'required parameters that could be undefined'. The reporter should _always_ be passed (it just happens to be undefined if there was an error creating it).

File: src/commands/createFunction.ts (hunk 1 above)
Comment (StephenWeatherford) [reply src/commands/createFunction.ts]: Cool.  Are we still guaranteed to receive the event even if you open a new workspace?

Comment (EricJizbaMSFT) [reply src/commands/createFunction.ts]: Not sure. I'll look into that separately: https://github.com/Microsoft/vscode-azurefunctions/issues/93

I don't want to create a separate telemetry event if I don't have to

**Pull request merged**